File: output/qwen_2.5_72b_instruct/avatar/direct_translation/python/java/codeforces_272_B.java

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class codeforces_272_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        scanner.nextLine();
        String[] input = scanner.nextLine().split(" ");
        int[] sequence = new int[input.length];
        for (int i = 0; i < input.length; i++) {
            sequence[i] = Integer.parseInt(input[i]);
        }
        System.out.println(nPairsWithCommonFX(sequence));
    }

    public static int nPairsWithCommonFX(int[] sequence) {
        Map<Integer, java.util.List<Integer>> storage = new HashMap<>();
        java.util.List<Integer> FXs = new java.util.ArrayList<>();
        for (int value : sequence) {
            int y = f(value);
            if (!storage.containsKey(y)) {
                storage.put(y, new java.util.ArrayList<>());
                storage.get(y).add(value);
                FXs.add(y);
            } else {
                storage.get(y).add(value);
            }
        }
        int sum = 0;
        for (int y : FXs) {
            int size = storage.get(y).size();
            sum += size * size - size;
        }
        return sum / 2;
    }

    public static int f(int n) {
        int y = 1;
        while (n != 1) {
            if (n % 2 != 0) {
                y += 1;
            }
            n /= 2;
        }
        return y;
    }
}

File: dataset/avatar/Python/Code/codeforces_272_B.py
def main ( ) :
    input ( )
    return nPairsWithCommonFX ( map ( int , input ( ).split ( ) ) )
def nPairsWithCommonFX ( sequence ) :
    storage = { }
    FXs = [ ]
    for value in sequence :
        y = f ( value )
        if y not in storage :
            storage [ y ] = [ value ]
            FXs.append ( y )
        else :
            storage [ y ].append ( value )
    return ( sum ( len ( storage [ y ] ) * len ( storage [ y ] ) for y in FXs ) - sum ( len ( storage [ y ] ) for y in FXs ) ) // 2
def f ( n ) :
    y = 1
    while n != 1 :
        if n % 2 :
            y += 1
        n //= 2
    return y
if __name__ == '__main__' :
    print ( main ( ) )

